uplink/ecclient/client.go

```diff
@@ -494,10 +494,7 @@ type clientCloser struct {
 }
 
 func (client *clientCloser) Close() error {
-	return errs.Combine(
-		client.Downloader.Close(),
-		client.client.Close(),
-	)
+	return client.client.Close()
```

Comment:
Maybe close Downloader via go routine to log the error and maybe prevent leaks?

Reply:
From the grpc go docs:
"Close tears down the ClientConn and all underlying connections. "

NewStream creates a new Stream for the client side. This is typically called by generated code. ctx is used for the lifetime of the stream.

To ensure resources are not leaked due to the stream returned, one of the following actions must be performed:

1. Call Close on the ClientConn.
2. Cancel the context provided.
3. Call RecvMsg until a non-nil error is returned. A protobuf-generated
   client-streaming RPC, for instance, might use the helper function
   CloseAndRecv (note that CloseSend does not Recv, therefore is not
   guaranteed to release all resources).
4. Receive a non-nil, non-io.EOF error from Header or SendMsg.


The main thing `Downloader.Close()` does is:
```
closeErr := client.stream.CloseSend()
_, recvErr := client.stream.Recv()
```

Which according to the documentation is not necessary if `ClientConn.Close()` is called

Reply:
well spoken.  :D

Reply:
We must still close the downloader, otherwise, we end up eventually leaking something, when we change Downloader. There's also a lot of error handling logic this is skipping.

https://github.com/storj/storj/blob/master/uplink/piecestore/download.go#L178

Reply:
@egonelbre What is your suggestion on this? Call `Downloader.Close()` in a new goroutine?

Reply:
I think if something was downloaded without interruption then we can skip closing `Downloader` but if something goes wrong before closing then with `Close` error we can get additional info what happened.

Reply:
Yes, skip `Recv` when we don't have an error pending.

Reply:
If we had proper sockets instead of streams, one option would be to close the underlying connection and then try to call Recv... or call Recv in a non-blocking manner... however grpc doesn't support either of them.